treyholt/Suzie
# Fixing the Dreaded Suzie Crash on URL changes

Ok so I will probably write a wiki on this later, but Suzie should actually finally be a lot more crash free from a bunch of different user interactions. 

At it's most basic I just continued to use the callback handler we had when we were notified that the url changed. 
However now I added a DOM polling mechanism. But it isn't enough to just do DOM polling, because for some reason, sometimes the DOM will load but the image element we need won't. So Instead it now polls on exactly the elements we need on an interval of every 100 ms. 


Basically once we see the url has changed, we check every 100 ms to see if the html elements we need have been properly laid out. We also only perform this check when we know we're on a valid URL for a top. 
Only when those two conditions are valid do we then go to extract the html items and ask for a recommendation. 

**To Test**
Clone the branch
As a reminder, in terminal, do: 
`git checkout main`
`  git pull`
  `git checkout th/LifeCycleBugFixes`

Refresh Suzie in the Chrome Settings Extension Page

And then try refreshing while looking at a top, navigating all around and looking at a top. Go back to a top, go click on a non-top, just try everything and make sure the recs are changing appropriately and Suzie is disappearing when it should.
